### ISSUE TITLE
how to use getMedia in controller with my another database table or own model

### ISSUE THREAD

**Issue by mansoormilad** [spatie/laravel-medialibrary#1850]

**Comment by acontia** [comment]
Same question here.

I have created my own model as described here: https://docs.spatie.be/laravel-medialibrary/v8/advanced-usage/using-your-own-model/
so, I have:
```
<?php

namespace App;

use Spatie\MediaLibrary\MediaCollections\Models\Media as BaseMedia;

class Media extends BaseMedia
{

}
```

I also have an Item model on which I want to attach images:
```
class Item extends Model implements HasMedia
{
    use InteractsWithMedia;
...
```

On my `ItemController` `show` function, I'm trying to get the main image URL.
I was expecting this to work, but it returns NULL:
`$item->getFirstMediaUrl()`

PHPStorm autocompletes this correctly, so seems like the function `getFirstMediaUrl` should be avaiable on `$item`.

However this works:
`$item['media'][0]->getFullUrl()`

Any idea of what can be wrong here?

**Comment by acontia** [comment]
Same question here.

I have created my own model as described here: https://docs.spatie.be/laravel-medialibrary/v8/advanced-usage/using-your-own-model/
so, I have:
```
<?php

namespace App;

use Spatie\MediaLibrary\MediaCollections\Models\Media as BaseMedia;

class Media extends BaseMedia
{

}
```

and
```
// config/media-library.php
...
    /*
     * The fully qualified class name of the media model.
     */
    'media_model' => App\Media::class,
...
```

I also have an Item model on which I want to attach images:
```
class Item extends Model implements HasMedia
{
    use InteractsWithMedia;
...
```

On my `ItemController` `show` function, I'm trying to get the main image URL.

I was expecting this to work, but it returns NULL:
`$item->getFirstMediaUrl()`

PHPStorm autocompletes this correctly, so seems like the function `getFirstMediaUrl` should be avaiable on `$item`.

I have found as workaround that this works, but seems wrong to me:
`$item['media'][0]->getFullUrl()`

Any idea on how to get this working?
